mmn/mmn.go

```diff
@@ -0,0 +1,77 @@
+package mmn
+
+import (
+	"github.com/mysteriumnetwork/node/core/connection"
+	"github.com/mysteriumnetwork/node/core/service"
+	"github.com/mysteriumnetwork/node/eventbus"
+	"github.com/mysteriumnetwork/node/identity"
+
+	"github.com/rs/zerolog/log"
+)
+
+type MMN struct {
+	collector *Collector
+	client    *client
+	eventBus  eventbus.EventBus
+}
+
+func NewMMN(collector *Collector, client *client, eventBus eventbus.EventBus) *MMN {
+	return &MMN{collector, client, eventBus}
+}
+
+func (m *MMN) SubscribeToEvents() {
+	m.subscribeToRegistrationEvent()
+	m.subscribeToProviderEvent()
+	m.subscribeToConsumerEvent()
+}
+
+func (m *MMN) Register(identity string) error {
+	m.collector.SetIdentity(identity)
+
+	return m.client.RegisterNode(m.collector.GetCollectedInformation())
+}
+
+func (m *MMN) UpdateNodeType(isProvider bool) error {
+	// don't resend when node type is the same
+	if m.collector.GetNodeType() == isProvider {
+		return nil
+	}
+
+	m.collector.SetNodeType(isProvider)
+
+	return m.client.UpdateNodeType(m.collector.GetCollectedInformation())
+}
+
+func (m *MMN) subscribeToRegistrationEvent() {
+	m.eventBus.SubscribeAsync(
+		identity.IdentityUnlockTopic,
+		func(identity string) {
+			log.Debug().Msg("Registration event")
+
+			if err := m.Register(identity); err != nil {
+				log.Error().Msg("Failed to register to MMN: " + err.Error())
```

Comment:
Use Msgf which formats message error for you

```go
log.Error().Msgf("Failed to register to MMN: %v",  err)
```

Reply:
For errors, it should be: `log.Error().Err(err).Msg("Failed to register to MMN")` @ignasbernotas @anjmao